src/input_context/preset.rs

```diff
@@ -0,0 +1,149 @@
+use bevy::prelude::*;
+
+use crate::{Input, InputBind, InputCondition, InputModifier, Negate, SwizzleAxis};
+
+pub trait BindPreset {
+    fn bindings(self) -> impl Iterator<Item = InputBind>;
+
+    fn with_modifier<M: InputModifier + Clone>(self, modifier: M) -> PresetWithModifier<Self, M>
+    where
+        Self: Sized,
+    {
+        PresetWithModifier {
+            preset: self,
+            modifier,
+        }
+    }
+
+    fn with_condition<C: InputCondition + Clone>(self, condition: C) -> PresetWithCondition<Self, C>
+    where
+        Self: Sized,
+    {
+        PresetWithCondition {
+            preset: self,
+            condition,
+        }
+    }
+}
```

Comment:
These two methods could confuse users. For example, calling `with(GamepadStick::Left.with_modifier(DeadZone::default())`, won't work as expected. The only use case for them I could think of to is applying a modifier :thinking: 

We could fix the mentioned footgun by adding native support for bind groups. So instead of having only the input-level and action-level groups, we could allow input-level groups to be nested in order to share modifiers. This will also unlock a syntax like this:

```rust
ctx.bind::<DummyAction>()
    .with((Input::mouse_motion(), GamepadString::Right).with_modifier(DeadZone::default()));
```

But I think it's outside of the scope of the PR.

Reply:
Agreed, I was a bit skeptical of including these fns as well, but this can always be added later.